### PR TITLE
feat(Core/Common): delete inherited lib game-interface

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
##### CHANGES PROPOSED:
- Delete inherited lib `game-interface` from `common` library
- Part for dynamic linking

##### TESTS PERFORMED:
- Build-in Win 10 at VS 16.3.8

##### HOW TO TEST THE CHANGES:
1. build with `TOOLS`

##### KNOWN ISSUES AND TODO LIST:
- [x] Build in Linux

##### Target branch(es):
- [x] Master

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Winfidonarleyan** [comment]
Ready to test

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan travis build is failing

File: src/common/Collision/Management/MMapManager.cpp [hunk 1]
```diff
@@ -183,15 +185,14 @@ namespace MMAP
             return false;
         }
 
-        return false;
+        //return false;
```
Comment (FrancescoBorzi) [inline src/common/Collision/Management/MMapManager.cpp]: why not just remove it?

**Comment by Viste** [comment]
fresh Debian 10 VDS: https://paste2.org/EVV4zPOp

File: src/common/Collision/Management/MMapManager.cpp (hunk 1 above)
Comment (Winfidonarleyan) [reply src/common/Collision/Management/MMapManager.cpp]: Yes yes

**Pull request closed without merging**